### PR TITLE
[ray client][draft] Expose kill and cancel in client api stub

### LINKED ISSUE #15175: [ray client] Support public API: ray.cancel()

### LINKED ISSUE #15176: [ray client] Support public API: ray.kill()

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
## Related issue number

Closes https://github.com/ray-project/ray/issues/15175
Closes https://github.com/ray-project/ray/issues/15176

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

### CODE REVIEW

**Review by ericl — CHANGES_REQUESTED** [review]
Can you add unit tests?

File: python/ray/util/client/__init__.py [hunk 1]
```diff
@@ -1,8 +1,11 @@
-from typing import List, Tuple, Dict, Any
+from typing import List, Tuple, Dict, Any, TYPE_CHECKING
 from ray.job_config import JobConfig
 import os
 import sys
 import logging
+if TYPE_CHECKING:
+    from ray.util.client.common import ClientActorHandle
+    from ray.util.client.common import ClientObjectRef
```
Comment (AmeerHajAli) [inline python/ray/util/client/__init__.py:8]: why do you need this?

Comment (ijrsvt) [reply python/ray/util/client/__init__.py:8]: Is it okay to just import these?

Comment (DmitriGekhtman) [reply python/ray/util/client/__init__.py:8]: I think the import is necessary -- or at least my linter complains if I don't import.
As for the "if", I copy-pasted from here: https://github.com/ray-project/ray/blob/aa2f60b3d5ff3ac10278e0e007b19b6bee2ba066/python/ray/util/client/api.py#L6
Not sure why we need to check for `TYPE_CHECKING`, but might as well keep things consistent.

**Comment by DmitriGekhtman** [comment]
Ok, so looks like `kill` and `cancel` are already implemented. They are tested here:
https://github.com/ray-project/ray/blob/4289fa8d43dc4b38a0066cf54a0dadd84a658a77/python/ray/tests/test_client_terminate.py

`kill` and `cancel` are defined as methods of the `ClientAPI` class https://github.com/ray-project/ray/blob/4289fa8d43dc4b38a0066cf54a0dadd84a658a77/python/ray/util/client/api.py#L136
Every method of `ClientAPI` also becomes a method of `RayAPIStub` by means of the stub's `__get__attr__`
https://github.com/ray-project/ray/blob/4289fa8d43dc4b38a0066cf54a0dadd84a658a77/python/ray/util/client/__init__.py#L14 

The methods in `ClientAPI` are the ones that are supposed to be called after establishing a client connection. Hence the error message here: https://github.com/ray-project/ray/blob/4289fa8d43dc4b38a0066cf54a0dadd84a658a77/python/ray/util/client/__init__.py#L124 
and the comment here: 
https://github.com/ray-project/ray/blob/4289fa8d43dc4b38a0066cf54a0dadd84a658a77/python/ray/util/client/__init__.py#L110

**Comment by ijrsvt** [comment]
@DmitriGekhtman I'm still not able to get `cancel` or `kill` to work when using ray client

**Comment by DmitriGekhtman** [comment]
Noted! Will check it out, open an issue, and work an a fix.

**Comment by ijrsvt** [comment]
I think that is why #15182 does now work

**Pull request closed without merging**